OpenDreamRuntime.Tests/DMProject/code.dm

```diff
@@ -24,6 +24,82 @@
 		stack_overflow_test()
 	. = 2
 
-/world/New()
-	..()
-	world.log << "World loaded!"
+
+//
+// waitfor_1_a() should evaluate to 3
+//
+/world/proc/waitfor_1_a()
+	. = 1
+	. = waitfor_1_b()
+
+/world/proc/waitfor_1_b()
+	set waitfor = FALSE
+	. = 2
+	. = waitfor_1_c()
+	. = 3
+	sleep(1)
+	. = 4
+
+/world/proc/waitfor_1_c()
+	set waitfor = FALSE
+	. = 5
+	sleep(1)
+	. = 6
+
+//
+// waitfor_2_a should evaluate to 2
+//
+/world/proc/waitfor_2_a()
+	. = 1
+	. = waitfor_2_b()
+
+/world/proc/waitfor_2_b()
+	set waitfor = FALSE
+	. = 2
+	. = waitfor_2_c()
+	. = 3
+	sleep(1)
+	. = 4
+
+/world/proc/waitfor_2_c()
+	set waitfor = TRUE
+	. = 5
+	sleep(1)
+	. = 6
+
+//
+
+/world/proc/default_test(datum/arg = new())
+	return arg
+
+/world/proc/crazy_inferred_types()
```

Comment:
This isn't being used in any tests

Reply:
i was happy with this as long as it manages to compile (meaning that the new() is inferring _something_)